src/coreclr/jit/codegenxarch.cpp

```diff
@@ -3017,12 +3017,8 @@ void CodeGen::genCodeForCpBlkUnroll(GenTreeBlk* node)
 
         instruction simdMov = simdUnalignedMovIns();
 
-        unsigned regSize = XMM_REGSIZE_BYTES; // compiler->getSIMDVectorRegisterByteLength();
-
-        if (regSize == YMM_REGSIZE_BYTES && size < regSize)
-        {
-            regSize = XMM_REGSIZE_BYTES;
```

Comment:
did you find this optimization unneeded?

Reply:
Yes, Tanner's suggestion to not use `compiler->getSIMDVectorRegisterByteLength()` makes that check unnecessary. The new `regSize` initialization handles checking for the case where `size < YMM_REGSIZE_BYTES`.

Reply:
sorry, I thought it was optimization that for copy [0, 48) we generate  [0, 32), [32, 48) instead of [0, 32) [16, 48).
from https://github.com/dotnet/runtime/pull/55604#discussion_r669871127 discussion,
do you have it now or do you think it does not give us any wins?

Reply:
I don't expect any wins from doing [0, 32), [32, 48) vs [0, 32), [16, 48), I expect both to perform about equally for these inlined copies since the [0,32), [32, 48) case will still be using the same lower lanes of the SIMD register.

Reply:
Got it, thanks